src/app/ReadHandler.h

```diff
@@ -333,13 +332,14 @@ class ReadHandler : public Messaging::ExchangeDelegate
     bool IsIdle() const { return mState == HandlerState::Idle; }
 
     /// @brief Returns whether the ReadHandler is in a state where it can send a report and there is data to report.
-    bool IsReportable() const
+    bool ShouldStartReporting() const
```

Comment:
There's still some confustion here.

The old IsReportable, before any of the scheduler changes, was:

1. Always true for a Read-type handler.
2. Always true during priming for a Subscribe-type handler.

as long as they were in the "GeneratingReports" state.

So perhaps this logic should be something like:
```
  return  mState == HandlerState::GeneratingReports && (IsType(ReadHandler::InteractionType::Read) || IsPriming() || IsDirty());
```

And then it becomes obvious that in SetStateFlag the state flag change cannot affect the state if either we're a Read-type handler or priming, so we do not need the check and ScheduleRun codepath there.